src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -478,7 +463,7 @@ internal CompilationContext CreateCompilationContext(CSharpSyntaxNode syntax)
                 }
 
                 ImmutableArray<bool> dynamicFlags;
-                if (dynamicLocalMap.TryGetValue(i, out dynamicFlags))
+                if (dynamicLocalMap != null && dynamicLocalMap.TryGetValue(i, out dynamicFlags))
```

Comment:
They are null in the default MethodDebugInfo.  The alternative is to use ?? in the argument.